atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/keyvalue/api/cache/LockWatchValueScopingCacheImpl.java

```diff
@@ -171,6 +171,13 @@ public Void invalidateSome(Set<LockDescriptor> invalidatedLocks) {
         ensureStateRemoved(startTimestamp);
     }
 
+    @Override
+    public synchronized void dumpState() {
+        log.info("Dumping state from LockWatchValueScopingCacheImpl", SafeArg.of("currentVersion", currentVersion));
+        valueStore.dumpState();
+        snapshotStore.dumpState();
+    }
+
```

Comment:
Separately, I'd argue we should synchronize it in the interest of memory visibility. Otherwise, unless for some other tangential reason there's a happens-before relationship between actual LW operations and calls to the diagnostic endpoint, the diagnostic endpoint could read stale (or even some stale and some current - i.e., inconsistent) state!